domain/annotation/state/query.go

```diff
@@ -98,8 +98,24 @@ func uuidQueryForID(id annotations.ID) (string, sqlair.M, error) {
 	case annotations.KindStorage:
 		selector = "name"
 	case annotations.KindCharm:
-		kindName = "v_charm_url"
-		selector = "url"
+		// Special case the charm.
+		charmID, ok := id.Parts.(annotations.CharmID)
+		if !ok {
+			return "", sqlair.M{}, fmt.Errorf("%w for charm", annotationerrors.InvalidIDParts)
+		}
+
+		args := sqlair.M{
```

Comment:
And you could pass it straight in instead of casting and building this.

Reply:
But then you're tying a id to a database schema. That's crossing so may boundaries.